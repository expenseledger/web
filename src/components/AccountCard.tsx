import React from "react";
import { animated, config, useSpring } from "react-spring";
import { useRecoilValue } from "recoil";
import styled from "styled-components";
import { currencyState } from "../common/shareState";
import { formatNumber } from "../common/utils";

interface AccountCardProps {
    id: number;
    name: string;
    balance: number;
}

const Cards = [
    styled.div`
        background-color: #bdd4e7;
        background-image: linear-gradient(315deg, #bdd4e7 0%, #8693ab 74%);
    `,
    styled.div`
        background-color: #c7e9fb;
        background-image: linear-gradient(315deg, #c7e9fb 0%, #e61d8c 74%);
    `,
    styled.div`
        background-color: #89d4cf;
        background-image: linear-gradient(315deg, #89d4cf 0%, #6e45e1 74%);
    `,
    styled.div`
        background-color: #f9d29d;
        background-image: linear-gradient(315deg, #f9d29d 0%, #ffd8cb 74%);
    `,
    styled.div`
        background-color: #537895;
        background-image: linear-gradient(315deg, #537895 0%, #09203f 74%);
    `,
    styled.div`
        background-color: #89d8d3;
        background-image: linear-gradient(315deg, #89d8d3 0%, #03c8a8 74%);
    `,
];

const AccountCardComponent: React.FC<AccountCardProps> = (props) => {
    const { number } = useSpring({
        reset: false,
        from: { number: 0 },
        number: props.balance,
        delay: 200,
        config: config.default,
    });
    const currency = useRecoilValue(currencyState);
    const Card = Cards[props.id % Cards.length];

    return (
        <Card className="box has-text-right has-text-white p-5">
            <div className="is-size-3 has-text-weight-bold">{props.name}</div>
            <div className="is-size-5">
                <span>{currency}</span>
                <animated.span>{number.to((x) => formatNumber(x))}</animated.span>
            </div>
        </Card>
    );
};

const AccountCard = React.memo(AccountCardComponent);

export default AccountCard;
