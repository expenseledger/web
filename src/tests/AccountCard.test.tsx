import AccountCard from "../components/AccountCard";
import { render, screen } from "@testing-library/react";

describe("<AccountCard>", () => {
    it("renders without crashing", () => {
        const { container } = render(
            <AccountCard id={1} balance={12.34} name="test" currency="à¸¿" />
        );

        expect(container).toMatchSnapshot();
    });
});
