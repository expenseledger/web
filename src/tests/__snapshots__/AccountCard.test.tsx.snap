// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountCard> renders without crashing 1`] = `
<div>
  <div
    class="sc-blHHSb kwWOcI"
  >
    <span
      class="rt-Text rt-r-size-8 rt-r-weight-bold"
    >
      test
    </span>
    <div
      class="rt-Box"
    >
      <span
        class="rt-Text rt-r-size-4"
      >
        à¸¿
      </span>
      <span
        class="rt-Text rt-r-size-4"
      >
        <span>
          0
        </span>
      </span>
    </div>
  </div>
</div>
`;
